migrations/lib/importer/steps/posts/posts.rb

```diff
@@ -0,0 +1,204 @@
+    end
+
+    def transform_row(row)
+      topic_id = row[:discourse_topic_id]
+
+      row[:topic_id] = topic_id
+      row[:user_id] = row[:discourse_user_id] || SYSTEM_USER_ID
+      row[:reply_to_user_id] = row[:discourse_reply_to_user_id]
+      row[:last_editor_id] = row[:discourse_last_editor_id]
+      row[:deleted_by_id] = row[:discourse_deleted_by_user_id]
+      row[:locked_by_id] = row[:discourse_locked_by_user_id]
+
+      existing_max_post_number = @max_post_number_by_topic_id[topic_id]
+      row[:post_number] = existing_max_post_number + row[:computed_post_number]
+
+      # TODO(selase): Handle replies to existing posts
+      if row[:reply_to_post_number]
+        row[:reply_to_post_number] = existing_max_post_number + row[:reply_to_post_number]
+      end
+
+      row[:post_type] = ensure_valid_value(
+        value: row[:post_type],
+        allowed_set: POST_TYPES,
+        default_value: DEFAULT_POST_TYPE,
+      ) do |value, default_value|
+        puts "   #{row[:original_id]}: Post type '#{value}' is invalid, defaulting to #{default_value}"
+      end
+
+      row[:hidden] = false if row[:hidden].nil?
+      row[:user_deleted] = false if row[:user_deleted].nil?
+      row[:wiki] = false if row[:wiki].nil?
+
+      if row[:hidden]
+        row[:hidden_reason_id] = ensure_valid_value(
+          value: row[:hidden_reason_id],
+          allowed_set: HIDDEN_REASONS,
+          default_value: nil,
+        )
+
+        row[:hidden_at] ||= row[:updated_at] || row[:created_at]
+      else
+        row[:hidden_reason_id] = nil
+        row[:hidden_at] = nil
+      end
+
+      row[:raw] = clean_raw(row[:raw], row[:post_type])
+      if row[:raw] == BLANK_POST_RAW || row[:raw] == EMPTY_POST_RAW
+        row[:word_count] = 0
+      else
+        row[:word_count] = row[:raw].scan(/[[:word:]]+/).size
+      end
+
+      # TODO(selase): Incomplete, temporary stand-in until we have a parser
+      row[:cooked] = PrettyText.cook(row[:raw])
+      row[:like_count] ||= 0
+      row[:last_version_at] ||= row[:created_at]
+
+      super
+    end
+
+    def clean_raw(raw, post_type)
+      if raw
+        raw.scrub!
+        raw.strip!
+      end
+
+      return raw if raw.present?
+
+      if post_type == SMALL_ACTION_TYPE || post_type == MODERATOR_ACTION_TYPE
+        BLANK_POST_RAW
+      else
+        EMPTY_POST_RAW
+      end
+    end
+
+    def load_existing_max_post_numbers
+      @max_post_number_by_topic_id = Hash.new(0)
+
+      topic_ids = @intermediate_db.query_splat <<~SQL, MappingType::TOPICS
+        SELECT DISTINCT mapped_topic.discourse_id
+          FROM posts
+               JOIN mapped.ids mapped_topic
+                 ON posts.topic_id = mapped_topic.original_id AND mapped_topic.type = ?
+      SQL
+
+      return if topic_ids.empty?
+
+      DB
+        .query(<<~SQL, topic_ids)
+        SELECT topic_id, COALESCE(MAX(post_number), 0) AS max_post_number
+        FROM posts
+        WHERE topic_id IN (?)
```

Comment:
It's very likely that a huge amount of values in `IN` will be causing problems.  What if we stored the `post_number` during the import in `mapped.posts`? I feel like we are going to need it anyway. If we resume an import, we'd simply calculate the max for each topic.

Reply:
Good, let's make it so.